hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ratis/utils/OzoneManagerRatisUtils.java

```diff
@@ -437,4 +450,117 @@ private static ServiceException createLeaderNotReadyException(
     return new ServiceException(leaderNotReadyException);
   }
 
+  public static boolean isBucketFSOptimized(OMRequest omRequest,
+      OzoneManager ozoneManager) {
+    Type cmdType = omRequest.getCmdType();
+    BucketLayout bucketLayout = null;
+    OmBucketInfo buckInfo = null;
+    switch (cmdType) {
+    case AllocateBlock:
```

Comment:
How about avoiding these duplicate switch statement using following approach:

- Add new function which takes args like below:

```
  private static boolean isBucketFSOptimized(String volName, String buckName,
      OzoneManager ozoneManager) {
    BucketLayout bucketLayout = null;
    OmBucketInfo buckInfo = null;
    buckInfo = getOmBucketInfo(ozoneManager, buckInfo, volName, buckName);
    bucketLayout = getBucketLayout(buckInfo);
    return BucketLayout.FILE_SYSTEM_OPTIMIZED.equals(bucketLayout);
  }
```
- Call the above function from the existing switch-case like below. The same can be repeated to other bucket layout cases as well : CreateKey, CommitKey etc 

```
    case AllocateBlock:
      keyArgs = omRequest.getAllocateBlockRequest().getKeyArgs();
      if (isBucketFSOptimized(keyArgs.getVolumeName(), keyArgs.getBucketName(),
          ozoneManager)) {
        return new OMAllocateBlockRequestWithFSO(omRequest);
      }
      return new OMAllocateBlockRequest(omRequest);

    case CreateKey:
      keyArgs = omRequest.getCreateKeyRequest().getKeyArgs();
      if (isBucketFSOptimized(keyArgs.getVolumeName(), keyArgs.getBucketName(),
          ozoneManager)) {
        return new OMKeyCreateRequestWithFSO(omRequest);
      }
      return new OMKeyCreateRequest(omRequest);

.....
....
```

Reply:
Thanks, @rakeshadr for this nice approach, I have made the necessary changes.